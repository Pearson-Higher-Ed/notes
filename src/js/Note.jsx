import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import SvgIcon from 'material-ui/SvgIcon';


export default class Note extends React.Component {
  constructor(props) {
    super(props);

    this.handleNoteClick = this.handleNoteClick.bind(this);
    this.handleDeleteClick = this.handleDeleteClick.bind(this);

    this.state = {
      modalOpen: false
    };
  }

  handleModalOpen = () =>  {
    this.setState({modalOpen: true});
  };

  handleModalClose = () =>  {
    this.setState({modalOpen: false});
  };

  handleNoteClick(pageId) {
   // window.pubsub.publish('GO_TO_PAGE', pageId);
  }

  handleDeleteClick(noteId) {
    this.props.removeNote(noteId);
    this.handleModalClose();
  }

  renderIcon(commentExists) {
    if (commentExists) {
      return(
        <i className={this.props.color.toLowerCase() + '-icon color-icon'}>
                <span className="reader-only">{this.props.color} Highlight</span>
        </i>
      )
    }
    else {
      return (
        <span className={this.props.color + ' color-dot'}>
            <span className="reader-only">
                {this.props.color} Highlight
            </span>
        </span>
      )
    }
  }
  onFocus(e) {
    e.target.parentNode.classList.add('focused');
    return true;
  }
  onBlur(e) {
    e.target.parentNode.className = 'note-row';
    return true;
  }
  handleUp(e) {
    switch(e.key) {
      case "ArrowDown":
          this.props.handleFocus(this.props.id+1);
      break;
      case "ArrowUp":
          this.props.handleFocus(this.props.id-1); 
      break;
    }
  }

  render() {
    const that = this;
    // const {formatDate} = this.props.translations;
    const commentExists = (this.props.comment) ? true : false;

    const DialogStyle = {
      dialogContainerstyl : {
        width: '362px',
        backgroundColor: '#fff',
        borderRadius: '4px'
      },
      cancelIcon: {
        color: '#8d8d8d',
        position: 'absolute',
        top: '19px',
        right: '19px',
        height: '18.7px',
        width: '18px',
        cursor: 'pointer'
      },
      cancelbtnstyl : {
        color: '#74797b'
      },
      deleteBtnstyl : {
        borderRadius: 2,
        backgroundColor: '#34b6b4'
      }
    }
    const actions = [
      <FlatButton
        label="Cancel"
        primary={true}
        style={DialogStyle.cancelbtnstyl}
        className="cancelBtn"
        onClick={this.handleModalClose} />,
      <FlatButton
        label="Delete"
        primary={true}
        onClick={this.handleDeleteClick.bind(this, this.props.id)}
        style={DialogStyle.deleteBtnstyl}
        className="deleteBtn" />
    ];
    const CancelIcon = (props) => (
      <SvgIcon {...props}>
        <path d="M712.993036,23.3253012 L720.736289,15.2808193 C721.026024,14.9878313 721.026024,14.5128916 720.735855,14.2199036 C720.445687,13.9266988 719.975518,13.9266988 719.685349,14.2199036 L711.976795,22.005012 L704.268458,14.2199036 C703.978072,13.9266988 703.50812,13.9266988 703.217735,14.2199036 C702.927566,14.5128916 702.927566,14.9878313 703.217518,15.2808193 L710.960554,23.3253012 L703.217735,31.3697831 C702.927566,31.6627711 702.927566,32.1377108 703.217735,32.4306988 C703.507687,32.7234699 703.978072,32.7239036 704.268458,32.4306988 L711.976795,24.6455904 L719.685349,32.4306988 C719.975518,32.7239036 720.44612,32.7234699 720.735855,32.4306988 C721.026024,32.1377108 721.026024,31.6627711 720.735855,31.3697831 L712.993036,23.3253012 Z" id="ic_cancel" fill="#8d8d8d"></path>
      </SvgIcon>
    );

    return (
      <div className="note-row">
          <a href="javascript:void(0);"
             className="note-link"
             onFocus= {that.onFocus.bind(that)}
             tabIndex="0"
             onKeyUp={this.handleUp.bind(this)}
             ref='a'
             onClick={this.handleNoteClick.bind(that, this.props.pageId)}>
              {this.renderIcon(commentExists)}
              <div className="note-content">
                  <div>
                      <p className="dotdotdot">
                          <strong>{this.props.text}</strong>
                      </p>
                  </div>
                  <div className={commentExists ? '' : 'hide'}>
                      <p className="user-note">
                          {this.props.comment}
                      </p>
                  </div>
                  <span className="note-date">
                    <time value={this.props.time} >{this.props.time}</time>
                  </span>
              </div>
          </a>
          <a href="javascript:void(0);"
             className="remove"
             tabIndex="0"
             onBlur={that.onBlur.bind(that)}
             onClick={this.handleModalOpen}
             aria-label="Remove highlight">
          </a>
          <Dialog
            title="Confirm Delete?"
            actions={actions}
            modal={false}
            open={this.state.modalOpen}
            onRequestClose={this.handleModalClose}
            contentStyle={DialogStyle.dialogContainerstyl}>
              <CancelIcon onClick={this.handleModalClose} viewBox="703 14 18 18.7" style={DialogStyle.cancelIcon} />
              This action cannot be undone.
          </Dialog>
      </div>
    )
  }
}