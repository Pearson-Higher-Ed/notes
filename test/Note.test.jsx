/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/

/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import Note from '../src/js/Note';
import msgObject from '../translations';

const intlObj = new InternationalSupport(msgObject, 'en');

// const NoteComp = injectIntl(({date, intl}) => (
//     <Note key={1}
//     		intl={intl}
//         id={123}
//         author='author'
//         title='title'
//         description='description'
//         />
// ));

it('renders correctly', () => {
  const tree = renderer
    .create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><Note /></IntlProvider></MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

	it('renders the Note List', () => {
  	const component = renderer.create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><Note key='123'
          id='123'
          author='note.author}'
          pageId='{note.pageId}'
          color='{note.color}'
          comment='{note.comment}'
          text='{note.text}'
          time='{note.time}'
          removeNote={function(){}}
          noteClick={function(){}}
          /></IntlProvider></MuiThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
 	
 	console.log(tree.children[0].props);

 // tree.children[0].props.onFocus();
  tree.children[0].props.onClick({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
  tree.children[0].props.onFocus({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
  tree.children[0].props.onKeyUp({preventDefault(){}, which:37, keyCode:37, target:{parentNode:{classList:{remove(){}}}}});
  //tree.children[1].props.onClick();

  // re-rendering
  // tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});



