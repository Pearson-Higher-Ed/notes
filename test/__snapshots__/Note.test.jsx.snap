// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="note-row"
>
  <a
    className="note-link"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyUp={[Function]}
    role="link"
    tabIndex="0"
  >
    <span
      className="undefined color-dot"
    >
      <span
        className="reader-only"
      >
         Highlight
      </span>
    </span>
    <div
      className="note-content"
    >
      <div>
        <p
          className="dotdotdot"
        >
          <strong />
        </p>
      </div>
      <div
        className="hide"
      >
        <p
          className="user-note"
        />
      </div>
      <span
        className="note-date"
      >
        <time
          value={undefined}
        >
          Invalid Date
        </time>
         
        <time
          value={undefined}
        >
          Invalid Date
        </time>
      </span>
    </div>
  </a>
  <a
    aria-label="Remove note"
    className="remove"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;

exports[`renders the Note List 1`] = `
<div
  className="note-row"
>
  <a
    className="note-link"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyUp={[Function]}
    role="link"
    tabIndex="0"
  >
    <i
      className="{note.color}-icon color-icon"
    >
      <span
        className="reader-only"
      >
        {note.color}
         Highlight
      </span>
    </i>
    <div
      className="note-content"
    >
      <div>
        <p
          className="dotdotdot"
        >
          <strong>
            {note.text}
          </strong>
        </p>
      </div>
      <div
        className=""
      >
        <p
          className="user-note"
        >
          {note.comment}
        </p>
      </div>
      <span
        className="note-date"
      >
        <time
          value="{note.time}"
        >
          Invalid Date
        </time>
         
        <time
          value="{note.time}"
        >
          Invalid Date
        </time>
      </span>
    </div>
  </a>
  <a
    aria-label="Remove note"
    className="remove"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;

exports[`renders the Note List 2`] = `
<div
  className="note-row"
>
  <a
    className="note-link"
    onClick={[Function]}
    onFocus={[Function]}
    onKeyUp={[Function]}
    role="link"
    tabIndex="0"
  >
    <span
      className="undefined color-dot"
    >
      <span
        className="reader-only"
      >
         Highlight
      </span>
    </span>
    <div
      className="note-content"
    >
      <div>
        <p
          className="dotdotdot"
        >
          <strong />
        </p>
      </div>
      <div
        className="hide"
      >
        <p
          className="user-note"
        />
      </div>
      <span
        className="note-date"
      >
        <time
          value={undefined}
        >
          Invalid Date
        </time>
         
        <time
          value={undefined}
        >
          Invalid Date
        </time>
      </span>
    </div>
  </a>
  <a
    aria-label="Remove note"
    className="remove"
    onClick={[Function]}
    role="link"
    tabIndex="0"
  >
    
  </a>
</div>
`;
